src/charm.py

```diff
@@ -954,6 +976,70 @@ def _core_reconciliation(self):
         if self._current_wp_config() is not None:
             self._start_server()
 
+    def _wp_theme_list(self):
+        """List all installed WordPress themes"""
+        process = self._run_wp_cli(["wp", "theme", "list", "--format=json"], timeout=600)
+        if process.return_code != 0:
+            raise RuntimeError("wp theme list command failed: " + process.stderr)
+        try:
+            return json.loads(process.stdout)
+        except json.decoder.JSONDecodeError:
+            raise RuntimeError(
+                "wp theme list command failed, stdout is not json: " + repr(process.stdout)
+            )
+
+    def _wp_theme_install(self, theme):
+        """Install WordPress theme"""
+        process = self._run_wp_cli(
+            # --force will overwrite any installed version of the theme,
+            # without prompting for confirmation
+            ["wp", "theme", "install", theme, "--force"],
+            timeout=600,
+            combine_stderr=True
+        )
+        if process.return_code != 0:
+            raise RuntimeError("wp theme list command failed: " + process.stdout)
+
+    def _wp_theme_delete(self, theme):
+        """Uninstall WordPress theme"""
+        process = self._run_wp_cli(
+            ["wp", "theme", "delete", theme, "--force"],
+            combine_stderr=True
+        )
+        if process.return_code != 0:
+            raise RuntimeError("wp theme list command failed: " + process.stdout)
+
+    def _theme_reconciliation(self):
+        """Reconciliation process for WordPress themes
+
+        Install and uninstall themes to match the themes setting in config
+        """
+        logger.debug("Start theme reconciliation process")
+        current_installed_themes = set(t["name"] for t in self._wp_theme_list())
+        logger.debug("Currently installed themes %s", current_installed_themes)
+        themes_in_config = [t.strip() for t in self.model.config["themes"].split(",")]
```

Comment:
What about  `themes_in_config = [t.strip() for t in self.model.config["themes"].split(",")] if self.model.config["themes"].strip() else []`? Would mean the following `if` statement isn't necessary

Reply:
I came up with a even better way to do this
```
themes_in_config = [
    t.strip() for t in self.model.config["themes"].split(",")
    if t.strip()
]
```